src/BenchmarkDotNet/EventProcessors/CompositeEventProcessor.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Toolchains.Results;
+using BenchmarkDotNet.Validators;
+
+namespace BenchmarkDotNet.EventProcessors
+{
+    internal sealed class CompositeEventProcessor : EventProcessorBase
+    {
+        private readonly IReadOnlyCollection<EventProcessorBase> eventProcessors;
```

Comment:
We should prefer concrete types over abstractions for private fields (https://github.com/dotnet/roslyn-analyzers/pull/6370)

```suggestion
        private readonly HashSet<EventProcessorBase> eventProcessors;
```